src/parser/warrior/fury/modules/azerite/UnbridledFerocity.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import Analyzer from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import { formatNumber, formatPercentage, formatThousands } from 'common/format';
+import { calculateAzeriteEffects } from 'common/stats';
+import calculateBonusAzeriteDamage from 'parser/core/calculateBonusAzeriteDamage';
+import TraitStatisticBox from 'interface/others/TraitStatisticBox';
+import SPELLS from 'common/SPELLS';
+import { SELECTED_PLAYER } from 'parser/core/EventFilter';
+import RAMPAGE_COEFFICIENTS from '../spells/RAMPAGE_COEFFICIENTS.js';
+
+const RAMPAGE = [SPELLS.RAMPAGE_1, SPELLS.RAMPAGE_2, SPELLS.RAMPAGE_3, SPELLS.RAMPAGE_4];
+const RECKLESSNESS_DURATION = 10000;
+
+class UnbridledFerocity extends Analyzer {
+  static dependencies = {
+    statTracker: StatTracker,
+  };
+
+  totalDamage = 0;
+  recklessnessEvents = [];
+
+  constructor(...args) {
+    super(...args);
+
+    this.active = this.selectedCombatant.hasTrait(SPELLS.UNBRIDLED_FEROCITY.id);
+    
+    if(!this.active) {
+      return;
+    }
+
+    this.traitBonus = this.selectedCombatant.traitsBySpellId[SPELLS.UNBRIDLED_FEROCITY.id].reduce((total, rank) => {
+      const [ damage ] = calculateAzeriteEffects(SPELLS.UNBRIDLED_FEROCITY.id, rank);
+      return total + damage;
+    }, 0);
+
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(RAMPAGE), this.onTraitDamage);
+    this.addEventListener(Events.applybuff.to(SELECTED_PLAYER).spell(SPELLS.RECKLESSNESS), this.onBuffApply);
+    this.addEventListener(Events.removebuff.to(SELECTED_PLAYER).spell(SPELLS.RECKLESSNESS), this.onBuffRemoved);
+    this.addEventListener(Events.fightend, this.onFightEnd);
+  }
+
+  onTraitDamage(event) {
+    const coefficient = RAMPAGE_COEFFICIENTS.find(r => r.id === event.ability.guid).coefficient;
+    const [ bonusDamage ] = calculateBonusAzeriteDamage(event, [this.traitBonus], coefficient, this.statTracker.currentStrengthRating);
+    this.totalDamage += bonusDamage;
+  }
+
+  onBuffApply(event) {
+    this.recklessnessEvents.push({
+      start: event.timestamp,
+      end: null,
+    });
+  }
+
+  onBuffRemoved(event) {
+    const latestRecklessness = this.recklessnessEvents.pop();
+    latestRecklessness.end = event.timestamp;
+    this.recklessnessEvents.push(latestRecklessness);
+  }
+
+  onFightEnd(event) {
+    const latestRecklessness = this.recklessnessEvents.pop();
+    if(!latestRecklessness.end) {
+      latestRecklessness.end = event.timestamp;
+    }
+    this.recklessnessEvents.push(latestRecklessness);
+  }
+
+  get damagePercentage() {
+    return this.owner.getPercentageOfTotalDamageDone(this.totalDamage);
+  }
+
+  get increasedRecklessnessDuration() {
+    return this.recklessnessEvents.reduce((total, event) => {
+      const duration = event.end - event.start;
+      return total + (duration > RECKLESSNESS_DURATION ? duration - RECKLESSNESS_DURATION : duration);
```

Comment:
I don't quite understand the logic here - if you had over 10s of the buff duration, it only counts the bonus duration, but otherwise the whole duration? Also, it doesn't seem you distinguish the normal Recklessness (https://www.wowhead.com/spell=1719/recklessness) from the one gained via this trait? 

Some clarification please 😄

Reply:
Wait, probably nevermind - the Recklessness buff has 10 sec duration, so if it's more than that, the trait procced sometimes during that.. And if it's less than 10 sec, then it's the trait again and you count the duration as a whole, correct? 

In which case, a comment explaining this would be nice. Also, you might want to add some kind of buffer, I doubt the actual 10 second buff would last exactly 10000ms, but in reality probably few ms more or less (which means a "trait" proc that should last for 4 seconds could last for let's say 9.85 seconds)